src/main/java/fashionable/simba/yanawaserver/matching/domain/service/ParticipationService.java

```diff
@@ -0,0 +1,58 @@
+
+import fashionable.simba.yanawaserver.matching.constant.ParticipationStatusType;
+import fashionable.simba.yanawaserver.matching.constant.RecruitmentStatusType;
+import fashionable.simba.yanawaserver.matching.domain.Participation;
+import fashionable.simba.yanawaserver.matching.domain.Recruitment;
+import fashionable.simba.yanawaserver.matching.domain.repository.ParticipationRepository;
+import fashionable.simba.yanawaserver.matching.domain.repository.RecruitmentRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class ParticipationService {
+    private final ParticipationRepository participationRepository;
+    private final RecruitmentRepository recruitmentRepository;
+
+    public ParticipationService(ParticipationRepository participationRepository, RecruitmentRepository recruitmentRepository) {
+        this.participationRepository = participationRepository;
+        this.recruitmentRepository = recruitmentRepository;
+    }
+
+    public Participation createParticipation(Participation participation) {
+        Recruitment recruitment = recruitmentRepository.findRecruitmentById(participation.getRecruitmentId()).orElseThrow();
+        if (recruitment.getStatus() == RecruitmentStatusType.CLOSED) {
+            throw new IllegalArgumentException("모집이 종료되어 참가요청을 보낼 수 없습니다.");
+        }
+        Optional<Participation> beforeParticipation = participationRepository.findParticipationByUser(participation.getUserId(), participation.getRecruitmentId());
+        if (getParticipationStatus(beforeParticipation) == ParticipationStatusType.ACCEPTED
+                || getParticipationStatus(beforeParticipation) == ParticipationStatusType.REJECTED) {
+            throw new IllegalArgumentException("이전 참가요청에 대해 승인, 거절에 대하여 재요청을 보낼 수 없습니다.");
+        }
+        return participationRepository.save(participation);
+    }
+
+    public Participation acceptParticipation(Long participationId) {
+        Participation participation = participationRepository.findParticipationById(participationId).orElseThrow();
+        if(participation.getStatus() != ParticipationStatusType.WAITING) {
+            throw new IllegalArgumentException("참여요청이 대기중이 아니므로 참가승인을 할 수 없습니다.");
+        }
+        return participation.changeAcceptedParticipation();
+    }
+
+    public Participation rejectParticipation(Long participationId) {
+        Participation participation = participationRepository.findParticipationById(participationId).orElseThrow();
+        if(participation.getStatus() != ParticipationStatusType.WAITING) {
+            throw new IllegalArgumentException("참여요청이 대기중이 아니므로 참가거절을 할 수 없습니다.");
+        }
+        return participation.changeRejectedParticipation();
+    }
+
+    public ParticipationStatusType getParticipationStatus(Optional<Participation> participation) {
+        if(participation.isPresent()) {
+            return participation.orElseThrow().getStatus();
+        }
```

Comment:
이미 participation이 없으면 예외를 발생하라! 라는 메서드가 있으니 isPresent 메서드로 확인할 필요는 없어보입니다! 추가로 Optional로 인자로 받지 않는다면 해당 메서드는 스테이터스만 가져오라는 기능만하는 메서드가 될 것 같아요!

```suggestion
            return participation.orElseThrow().getStatus();
```